builtin/logical/ssh/path_roles.go

```diff
@@ -331,6 +333,17 @@ func pathRoles(b *backend) *framework.Path {
                                 If set, allows the enforcement of key types and minimum key sizes to be signed.
                                 `,
 			},
+			"algorithm_signer": &framework.FieldSchema{
+				Type:    framework.TypeString,
+				Default: ssh.SigAlgoRSA,
+				Description: `
+				When supplied, this value specifies a signing algorithm for the key.  Possible values: 
+				ssh-rsa, rsa-sha2-256, rsa-sha2-512.
```

Comment:
Probably needs a warning against ssh-rsa per our SHA1 warnings elsewhere, though possibly just in the docs.